### PR TITLE
[KAT-427] Provide Cython features useful for wrapping enterprise features.

### PR DESCRIPTION
**Pull request by arthurp**

### CODE REVIEW

File: python/katana/dynamic_bitset.pxd [hunk 1]
```diff
@@ -0,0 +1,5 @@
+from .cpp.libgalois.DynamicBitset cimport DynamicBitset as CDynamicBitset
+
```
Comment (gliga) [inline python/katana/dynamic_bitset.pxd:2]: not sure what changed here :)

File: python/katana/util/buffer_access.pyx [hunk 2]
```diff
@@ -0,0 +1,27 @@
+import numpy
+
+from cpython.buffer cimport PyBUF_CONTIG, PyBuffer_Release, PyObject_CheckBuffer, PyObject_GetBuffer
+
+
+cdef class UntypedBufferAccess:
+    def __init__(self, data):
+        # Handle pyarrow.Array
+        if not PyObject_CheckBuffer(data) and hasattr(data, "to_numpy"):
```
Comment (gliga) [inline python/katana/util/buffer_access.pyx]: why not move this check after the next if statement, then we do not need the first part of the condition

File: python/katana/util/buffer_access.pyx [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+
+        if not PyObject_CheckBuffer(data):
+            raise TypeError("data must support buffer protocol")
+
+        if hasattr(data, "dtype"):
+            self._numpy_dtype = data.dtype
+        else:
+            raise TypeError("data must provide a numpy dtype attribute")
+
+
```
Comment (gliga) [inline python/katana/util/buffer_access.pyx]: is there an extra line here

**Review by gliga — APPROVED** [review]

File: python/katana/dynamic_bitset.pxd (hunk 1 above)
Comment (arthurp) [reply python/katana/dynamic_bitset.pxd:2]: I needed to lift this code into a `.pxd` so I could call it from other Cython code directly.

**Pull request merged**